src/tiledb/cloud/file_ingestion/ingestion.py

```diff
@@ -0,0 +1,226 @@
+import os
+from math import ceil
+from typing import Any, List, Mapping, Optional, Sequence, Union
+
+import tiledb
+from tiledb.cloud import dag
+from tiledb.cloud.dag.mode import Mode
+from tiledb.cloud.utilities import as_batch
+from tiledb.cloud.utilities import get_logger_wrapper
+from tiledb.cloud.vcf.ingestion import find_uris_udf
+
+DEFAULT_RESOURCES = {"cpu": "1", "memory": "2Gi"}
+DEFAULT_FILE_INGESTION_NAME = "file-ingestion"
+DEFAULT_BATCH_SIZE = 100
+
+
+def sanitize_filename(fname: str) -> str:
+    """
+    Sanitizes a filename by removing invalid characters.
+
+    :param fname: A filename to sanitize
+    :return str: The sanitized string
+    """
+    return fname.replace(",", "").replace(" ", "_")
+
+
+def chunk_results_udf(
+    udf_results: Sequence[Sequence[str]],
+    batch_size: Optional[int] = DEFAULT_BATCH_SIZE,
+    verbose: bool = False,
+) -> List[List[str]]:
+    """
+    Flatten and break a list of udf results into batches of a specified size.
+
+    :param udf_results: An iterable of iterables containing UDF results.
+    :param batch_size: Resulting chunk size, defaults to 100.
+    :return List[List[str]]: A list of chunks as lists.
+    """
+    logger = get_logger_wrapper(verbose)
+
+    # Flatten the results into a list.
+    flattened_results = [result for udf_result in udf_results for result in udf_result]
+
+    # Reduce batch size if there are fewer sample URIs
+    batch_size = min(batch_size, len(flattened_results))
+    num_chunks = ceil(len(flattened_results) / batch_size)
+    logger.info(f"Splitting results into {num_chunks} chunks...")
+
+    return [
+        flattened_results[n * batch_size : (n + 1) * batch_size]
+        for n in range(num_chunks)
+    ]
+
+
+def ingest_files_udf(
+    dataset_uri: str,
+    file_uris: Sequence[str],
+    destination: str,
+    *,
+    acn: Optional[str] = None,
+    config: Optional[dict] = None,
+    namespace: Optional[str] = None,
+    verbose: bool = False,
+) -> None:
+    """
+    Ingest files.
+
+    :param dataset_uri: The dataset URI.
+    :param file_uris: An iterable of file URIs.
+    :param destination: URI to ingest the files into.
```

Comment:
Can `destination` be inferred by the `dataset_uri`? Assuming that `dataset_uri` is a group we can get its S3 path and use it for `destination`. If there is no `dataset_uri` we can write the files to the default storage location of the user.

Reply:
Current UI implementation provides a destination though.

Reply:
This can still be empty and get some default value from the provided group or default namespace

Reply:
The user should be required to pass the storage location. Let's keep all the ingestors consistent. All other ingestors have this value passed today.

Future improvements to all ingestors can be using defaults.